src/api/java/com/minecolonies/api/crafting/RecipeStorage.java

```diff
@@ -382,22 +383,29 @@ public boolean equals(final Object o)
     @Override
     public int hashCode()
     {
-        int result = cleanedInput.hashCode();
-        result = 31 * result + primaryOutput.hashCode();
-        result = 31 * result + (intermediate != null ? intermediate.hashCode() : 0);
-        result = 31 * result + gridSize;
-        if(recipeSource != null)
-        {
-            result = 31 * result + recipeSource.hashCode();
-        }
-        if(recipeType != null && !(recipeType instanceof ClassicRecipe))
+        return Objects.hash(cleanedInput, 
+            primaryOutput.getItem(), 
+            primaryOutput.getCount(), 
+            intermediate, 
+            gridSize, 
+            hashableItemStackList(alternateOutputs), 
+            hashableItemStackList(secondaryOutputs),
+            hashableItemStackList(tools));
+    }
+
+    /**
+     * Convert a list of itemstacks into something hashable
+     * @param items List of item stacks to convert
+     * @return hashtable of items and counts
+     */
+    private Map<Item, Integer> hashableItemStackList(List<ItemStack> items)
+    {
+        Map<Item, Integer> hashableList = new HashMap<>();
+        for(ItemStack item: items)
         {
-            result = 31 * result + recipeType.hashCode();
+            hashableList.put(item.getItem(), item.getCount());
         }
-
-        result = 31 * result + alternateOutputs.hashCode();
-        result = 31 * result + secondaryOutputs.hashCode();
-        return result;
+        return hashableList;
```

Comment:
Nope.

ItemStacks still need to be accurate.
See the JVM Spec on equals/hashCode.

Also to make this accurate, you need some combination of: Item, Damage and NBT.

Reply:
@OrionDevelopment we discussed this once already, we hash only item, like we do it in ItemStorage, and use equals for the fine tuned matching. That works very well.

Reply:
Why?
Like what is the reason todo that?

Reply:
Wouldn't it make more sense to use a `List<ItemStorage>` instead?  The main point of ItemStorage was to provide more sane equals/hash/large-count behaviour, I thought.

Reply:
I thought about that as well, but itemStorage doesn't include "count"